demos/common/cpp/models/include/models/model_base.h

```diff
@@ -17,23 +17,24 @@
 #pragma once
 #include "input_data.h"
 #include "results.h"
+#include "utils/config_factory.h"
 
 class ModelBase {
 public:
     ModelBase(const std::string& modelFileName)
-        : execNetwork(nullptr), modelFileName(modelFileName)
+        : modelFileName(modelFileName)
     {}
 
     virtual ~ModelBase() {}
 
-    virtual void prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNetwork) = 0;
     virtual std::shared_ptr<InternalModelData> preprocess(const InputData& inputData, InferenceEngine::InferRequest::Ptr& request) = 0;
     virtual std::unique_ptr<ResultBase> postprocess(InferenceResult& infResult) = 0;
-        virtual void onLoadCompleted(InferenceEngine::ExecutableNetwork* execNetwork, const std::vector<InferenceEngine::InferRequest::Ptr>& requests) {
-        this->execNetwork = execNetwork; }
+    virtual void onLoadCompleted(const std::vector<InferenceEngine::InferRequest::Ptr>& requests) {}
     const std::vector<std::string>& getOutputsNames() const { return outputsNames; }
     const std::vector<std::string>& getInputsNames() const { return inputsNames; }
 
+    virtual InferenceEngine::ExecutableNetwork loadExecutableNetwork(const CnnConfig& cnnConfig, InferenceEngine::Core& core);
+
     std::string getModelFileName() { return modelFileName; }
 
     virtual void reshape(InferenceEngine::CNNNetwork & cnnNetwork) {
```

Comment:
The name doesn't match the description. Rename or add custom shape setting.

Reply:
Done